shivyc/tree/expr_nodes.py

```diff
@@ -396,7 +404,19 @@ def _nonarith(self, left, right, il_code):
         raise CompilerError(err, self.op.r)
 
 
-class Div(_ArithBinOp):
+class _IntBinOp(_ArithBinOp):
+    """Base class for operations that works with integral type operands."""
+
+    def _check_type(self, left, right):
+        """Performs additional type check for operands.
+
+        left - ILValue for left operand
+        right - ILValue for right operand
+        """
+        return left.ctype.is_integral() and right.ctype.is_integral()
+
+
+class Div(_IntBinOp):
```

Comment:
I think Div should remain an _ArithBinOp because 6.5.5/2 says "Each of the operands shall have arithmetic type. The operands of the % operator shall have integer type." which seems to indicate division can have non-integer type.

Great fix on the Mod operator, though, and I really like your implementation of the bit shift nodes!